hw3/03-react/src/api/gotAPI.js

```diff
@@ -0,0 +1,18 @@
+
+/**
+ * (GET Request): Fetches data from the Game of Thrones API
+ * @param url the url path required to fetch the data
+ * @returns the response data as a .json() object
+ */
+const fetchData = async function fetchCharactersAPI() {
+  const url = "https://thronesapi.com/api/v2/Characters";
+  try {
+    const response = await fetch(url);
+    return await response.json();
+  } catch (error) {
+    throw new Error("Error fetching API data");
+  }
+};
```

Comment:
I would consider letting the user know if something goes wrong with the API call by adding a message to the DOM. As it stands right now, the loading gif will display indefinitely if something goes wrong with the API call.